<a href="https://colab.research.google.com/github/ashnaimran06/my-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ashnaimran06/my-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane and why

My lane is *Refresh/Content Opportunity Scoring*. I chose this lane because I want to find out which pages should be reviewed first for a possible content refresh. I am interested in using signals such as impressions, content age, freshness, position, CTR, and engagement to prioritize pages. This also builds on my earlier work with hand-written rules and decision trees, while allowing me to test whether a data-based approach can improve the ranking.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**My question is:** Which pages should be reviewed first for a possible content refresh?

An SEO or content editor would use the ranked list to decide which pages to review, update, improve, or monitor first.

A wrong recommendation could waste the editor's time on a page that does not need attention. On the other hand, missing an important page could mean that a page needing improvement is not reviewed in time.

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

stale = df["days_since_last_update"] >= 180
visible = df["impressions_90d"] >= 500

print("Stale pages:", stale.sum())
print("Visible pages:", visible.sum())
print("Stale + visible pages:", (stale & visible).sum())

Stale pages: 174
Visible pages: 16726
Stale + visible pages: 17


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

The starter dataset has 30,000 pages. Of these, 16,726 pages have at least 500 impressions, showing that many pages have enough visibility to be worth reviewing. There are also 174 stale pages, and 17 pages are both stale and highly visible. This makes content refresh prioritization a useful direction to investigate over the next 7 weeks.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

import pandas as pd

url = "https://raw.githubusercontent.com/ashnaimran06/my-ml-internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Total pages:", len(df))
print("Pages with 500+ impressions:", (df["impressions_90d"] >= 500).sum())
print("Pages stale (180+ days):", (df["days_since_last_update"] >= 180).sum())
print("Stale + visible pages:", (stale & visible).sum())

Total pages: 30000
Pages with 500+ impressions: 16726
Pages stale (180+ days): 174
Stale + visible pages: 17


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work will provide decision-support by identifying and ranking pages that may deserve a content review. I can make **observed and directional claims** about which content and search signals are associated with pages that look like refresh opportunities. I can compare different ranking methods and see which performs better on the available data. However, I cannot claim that refreshing a page will cause its performance to improve, and I cannot claim to predict Google's algorithm or prove why Google ranks a page in a certain position.

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.